usr/src/uts/common/fs/zfs/zfs_ioctl.c

```diff
@@ -2058,6 +2058,24 @@ zfs_ioc_objset_stats_impl(zfs_cmd_t *zc, objset_t *os)
 			error = zvol_get_stats(os, nv);
 			if (error == EIO)
 				return (error);
+			/*
+			 * If the zvol's parent dataset was being destroyed
```

Comment:
I don't understand this part of the comment.  If the zvol's parent dataset was being destroyed, then the parent must have no children (because parent can not be destroyed while it has children).  Therefore this zvol does not exist so this can't happen.  The dp_config_rwlock (which we hold for reader) prevents any race condition with dataset deletion.